Claude Code workflow fails on PRs from Copilot/external contributors

# Claude Code Workflow Fails on External PRs

## Problem

The Claude Code GitHub Action workflow fails on pull requests from GitHub Copilot (and likely any external contributor) with the error:

```
User does not have write access on this repository
```

## Error Details

From run [#19915955861](https://github.com/JacobPEvans/ai-assistant-instructions/actions/runs/19915955861):

```
App token exchange failed: 401 Unauthorized - User does not have write access on this repository
Attempt 3 failed: User does not have write access on this repository
Operation failed after 3 attempts
##[error]Failed to setup GitHub token: Error: User does not have write access on this repository

If you instead wish to use this action with a custom GitHub token or custom GitHub app, 
provide a `github_token` in the `uses` section of the app in your workflow yml file.
```

## Affected Branches

All `copilot/*` branches are failing:
- `copilot/implement-claude-sub-agents`
- `copilot/change-license-to-apache-2-0`
- `copilot/add-github-templates`
- `copilot/consolidate-duplicate-ai-commands`

## Root Cause

The `anthropics/claude-code-action@v1` uses OIDC token exchange to get a GitHub App token. This requires the PR author to have **write access** to the repository. When PRs come from:
- GitHub Copilot bot
- External contributors (forks)
- Users without write access

...the token exchange fails.

## Current Workflow

```yaml
# .github/workflows/claude.yml
- name: Run Claude Code Review
  uses: anthropics/claude-code-action@v1
  with:
    anthropic_api_key: ${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}
    # No github_token provided - relies on OIDC exchange
```

## Potential Solutions

### Option 1: Provide explicit github_token (Recommended)

```yaml
- name: Run Claude Code Review
  uses: anthropics/claude-code-action@v1
  with:
    anthropic_api_key: ${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}
    github_token: ${{ secrets.GITHUB_TOKEN }}
```

### Option 2: Skip for non-write users

Add a condition to skip the workflow for external contributors:

```yaml
jobs:
  claude:
    if: github.event.pull_request.head.repo.full_name == github.repository
    # ... rest of job
```

### Option 3: Use pull_request_target (Security risk)

```yaml
on:
  pull_request_target:
    types: [opened, synchronize]
```

⚠️ **Warning**: This has security implications as it runs with write access on untrusted code.

## Recommendation

**Option 1** is the safest - provide an explicit `github_token` so the action doesn't rely on OIDC token exchange.

## Action Items

- [ ] Update `.github/workflows/claude.yml` to include `github_token: ${{ secrets.GITHUB_TOKEN }}`
- [ ] Test on a new `copilot/*` PR
- [ ] Consider adding documentation about this limitation

## References

- [Anthropic Claude Code Action](https://github.com/anthropics/claude-code-action)
- [GitHub Actions: Automatic token authentication](https://docs.github.com/en/actions/security-guides/automatic-token-authentication)

--- a/.github/workflows/claude.yml
+++ b/.github/workflows/claude.yml
@@ -31,6 +31,8 @@ jobs:
         with:
           # OAuth token for Claude Code GitHub App authentication
           claude_code_oauth_token: ${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}
+          # Explicit GitHub token to bypass OIDC exchange (enables Copilot/bot PRs)
+          github_token: ${{ secrets.GITHUB_TOKEN }}
           track_progress: true
           use_sticky_comment: true
 

